### PR TITLE
Reflection context tests working

### PR DESCRIPTION
**Pull request by JosieBigler**
Contributes to https://github.com/dotnet/runtime/issues/26506

### CODE REVIEW

File: src/libraries/System.Reflection.Context/tests/ProjectingFieldInfoTests.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        {
+            TypeInfo typeInfo = typeof(SecondTestObject).GetTypeInfo();
+            _customTypeInfo = _customReflectionContext.MapType(typeInfo);
+            _field = _customTypeInfo.GetField("field");
+        }
+
+        [Fact]
+        public void DeclaringType_ReturnsCustomType()
+        {
+            Assert.Equal(ProjectionConstants.CustomType, _field.DeclaringType.FullName);
```
Comment (steveharter) [inline src/libraries/System.Reflection.Context/tests/ProjectingFieldInfoTests.cs:24]: This is one of the tests that is failing. It looks like DeclaringType is `SecondTestObject`.

Did this succeed for you (locally)?

Comment (JosieBigler) [reply src/libraries/System.Reflection.Context/tests/ProjectingFieldInfoTests.cs:24]: I've got myself setup again, and this is failing. For whatever reason, when i initially PR'd this i thought it was passing.  I'll button up the failures.

**Comment by JosieBigler** [comment]
@steveharter  I remember now, I had trouble hitting the `ProjectionFieldInfo` API. The Tests that I did do are in the PR.  These tests move the code coverage to just under 50%?

File: src/libraries/System.Reflection.Context/src/System/Reflection/Context/CustomReflectionContext.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using System.Reflection.Context.Custom;
 using System.Reflection.Context.Projection;
 using System.Reflection.Context.Virtual;
+using System.Runtime.CompilerServices;
```
Comment (steveharter) [inline src/libraries/System.Reflection.Context/src/System/Reflection/Context/CustomReflectionContext.cs:8]: I don't think you need this extra using.

Comment (JosieBigler) [reply src/libraries/System.Reflection.Context/src/System/Reflection/Context/CustomReflectionContext.cs]: Removed!

**Review by steveharter — APPROVED** [review]
Thanks!

**Comment by steveharter** [comment]
Mono tests failing not related:
```
Signing /tmp/helix/working/B95D0A64/w/A726092D/e/System.Runtime.Tests.app
    "/Users/helix-runner/Library/Keychains/signing-certs.keychain-db"
     0 valid identities found
No valid signing identities found in the keychain
```

**Pull request merged**